src/TestFramework/TestFramework/Attributes/DataSource/DynamicDataAttribute.cs

```diff
@@ -84,6 +84,12 @@ public DynamicDataAttribute(string dynamicDataSourceName, Type dynamicDataDeclar
     /// </summary>
     public Type? DynamicDataDisplayNameDeclaringType { get; set; }
 
+    /// <summary>
+    /// Gets or sets a value indicating whether the data source should be expanded, allowing to treat each data row as a separate test.
+    /// The default value is <c>true</c>.
+    /// </summary>
+    public bool ExpandDataSource { get; set; } = true;
```

Comment:
My feeling is for this to be `bool?`, and it means:

- null: Use what is specified by the assembly attribute and default to DuringDiscovery when the attribute is not present.
- true: force DuringDiscovery, even if the assembly attribute is set to DuringExecution
- false: force DuringExecution, even if the assembly attribute is set to DuringDiscovery

Reply:
Does it bring any advantage to consider the null state over simple true/false?

Reply:
@Evangelink If we are going to consider it in both ways (disable and enable), then there is no way for simple true/false to work, I think?

Reading the value as true will mean "use DuringDiscover", and reading the value as false will mean "use DuringExecution". But then nothing will mean "use whatever specified by the assembly attribute".

Reply:
Ah right, good point! I'll try to do the rework tonight